build-system/tasks/helpers.js

```diff
@@ -383,17 +440,35 @@ async function esbuildCompile(srcDir, srcFilename, destDir, options) {
       name: 'remap-dependencies',
       setup(build) {
         build.onResolve({filter: /.*/}, (args) => {
-          const {path: importPath, resolveDir} = args;
-          const dep = importPath.startsWith('.')
-            ? path.posix.join(resolveDir, importPath)
-            : importPath;
+          const {resolveDir} = args;
+          let {path: importPath} = args;
+
+          // Convert directory imports -> explicit imports of the index file.
+          // Leave js/ts extension out to be lang-agnostic.
+          if (importPath === './') {
+            importPath = './index';
+          }
+
+          let dep;
+          // Use resolvePath() the path to normalize files/directories.
+          // If file, gets filepath; if directory, gets the index filepath
+          if (importPath.startsWith('.')) {
+            const absPath = path.posix.join(resolveDir, importPath);
+            const rootDir = 'amphtml/';
+            const rootRelativePath = absPath.slice(
```

Comment:
I didn't test this but this seems like it would be easier to read as
`path.posix.relative(rootDir, absPath)`

Reply:
updated